servo-core/src/main/java/com/netflix/servo/tag/Tags.java

```diff
@@ -15,35 +15,22 @@
  */
 package com.netflix.servo.tag;
 
-import com.google.common.collect.Interner;
-import com.google.common.collect.Interners;
-
 /**
  * Helper functions for working with tags and tag lists.
  */
 public final class Tags {
-  /**
-   * Keep track of the strings that have been used for keys and values.
-   */
-  private static final Interner<String> STR_CACHE = Interners.newWeakInterner();
-
-  /**
-   * Keep track of tags that have been seen before and reuse.
-   */
-  private static final Interner<Tag> TAG_CACHE = Interners.newWeakInterner();
-
   /**
    * Intern strings used for tag keys or values.
    */
   public static String intern(String v) {
-    return STR_CACHE.intern(v);
+    return v.intern();
```

Comment:
This would have very different performance implications and could cause issues. The default string table in the JDK doesn't grow and can be quite small unless customized by the user and having too many interned strings can really degrade performance.

https://shipilev.net/jvm/anatomy-quarks/10-string-intern/

Reply:
Agreed, I was hoping we could take the performance hit for legacy consumers or if they notice the problem we can get them to update to spectator 😆